ee/indexeddb/indexeddb.go

```diff
@@ -0,0 +1,184 @@
+// Package indexeddb provides the ability to query an IndexedDB and parse the objects it contains.
+package indexeddb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/kolide/goleveldb/leveldb"
+	"github.com/kolide/goleveldb/leveldb/opt"
+	"github.com/kolide/launcher/ee/agent"
+)
+
+// maxObjectStoresToCheck is the number of indices for object stores we will check
+// before declaring failure to find the given object store. We cannot look up
+// object stores by their names, only by their IDs -- so we have to iterate through
+// up to maxObjectStoresToCheck IDs to find the desired store. We assume there are
+// fewer than 100 object stores in a given database.
+const maxObjectStoresToCheck = 100
+
+// QueryIndexeddbObjectStore queries the indexeddb at the given location `dbLocation`,
+// returning all objects in the given database that live in the given object store.
+func QueryIndexeddbObjectStore(dbLocation string, dbName string, objectStoreName string) ([]map[string]any, error) {
+	// If Chrome is open, we won't be able to open the db. So, copy it to a temporary location first.
+	tempDbCopyLocation, err := copyIndexeddb(dbLocation)
+	if err != nil {
+		if tempDbCopyLocation != "" {
+			_ = os.RemoveAll(tempDbCopyLocation)
+		}
+		return nil, fmt.Errorf("unable to copy db: %w", err)
+	}
+	defer os.RemoveAll(tempDbCopyLocation)
```

Comment:
Does this need a check on `tempDbCopyLocation != ""`? (I wouldn't normally think so, but there's one on line31)

Reply:
We don't need an additional check here -- we only get here if err == nil, and if err == nil the indexeddb was copied correctly to `tempDbCopyLocation`, so the location won't be an empty string.

The check on line 31 is to handle partial copy failures -- e.g. we made a directory and copied over a couple files, but not all of them, so we can't proceed but would still want to clean up that partial copy.